engine/src/main/java/com/datatorrent/stram/debug/TupleRecorder.java

```diff
@@ -463,10 +492,21 @@ private void publishTupleData(int portId, Object obj)
         wsClient.publish(recordingNameTopic, map);
       }
     } catch (Exception ex) {
-      logger.warn("Error publishing tuple data", ex);
+      if (shouldLog()) {
+        logger.warn("Error publishing tuple data {}", obj, ex);
+      }
     }
   }
 
+  private boolean shouldLog()
+  {
+    if ((lastLog == -1) || (totalTupleCount - lastLog) >= ERROR_LOG_GAP) {
```

Comment:
I thought you meant to set logGap to ERROR_LOG_GAP and decrement it till it reaches zero. In that case it would have to decremented for every tuple (after the first error) as the gap is the minimum number of tuples between errors and not number of errors in between. Maybe I am missing something here.